packages/shared/src/lib/core/layer-2/actions/fetchEvmBalancesForAccount.ts

```diff
@@ -2,14 +2,18 @@ import { IAccountState } from '@core/account/interfaces'
 import { getEvmNetworks } from '@core/network/stores'
 import { ITokenBalance } from '@core/token'
 import { setLayer2AccountBalanceForChain } from '@core/layer-2/stores'
+import { activeProfileId } from '@core/profile/stores'
+import { get } from 'svelte/store'
 
-export function fetchEvmBalancesForAccount(account: IAccountState): void {
+export function fetchEvmBalancesForAccount(profileId: string, account: IAccountState): void {
     const evmNetworks = getEvmNetworks()
     evmNetworks.forEach(async (evmNetwork) => {
         try {
             const tokenBalance: ITokenBalance = (await evmNetwork.getBalance(account)) ?? {}
 
-            setLayer2AccountBalanceForChain(account.index, evmNetwork.id, tokenBalance)
+            if (get(activeProfileId) === profileId) {
+                setLayer2AccountBalanceForChain(account.index, evmNetwork.id, tokenBalance)
+            }
         } catch (error) {
             console.error(error)
         }
```

Comment:
We should break out of the for loop if the activeProfileId changed.

Reply:
all of the loop iterations are fired off at the same time, so breaking out wouldnt do anything